lib/logstash/outputs/logstash.rb

```diff
@@ -169,7 +169,7 @@ def normalize_host_uris
   end
 
   def send_events(events)
-    body = events.map(&:to_json).map { |json| json << "\n" }.join
+    body = events.map(&:to_json).map { |json| json << "\n" }.reduce(:<<)
```

Comment:
Why do we need two maps, wouldn't that allocate extra intermediate array? 
```suggestion
events.map { |event| event.to_json << "\n" }
```